src/script/cpp_api/s_base.cpp

```diff
@@ -158,6 +160,34 @@ void ScriptApiBase::loadScript(const std::string &script_path)
 	lua_pop(L, 1); // Pop error handler
 }
 
+#ifndef SERVER
+void ScriptApiBase::loadModFromMemory(const std::string &mod_name)
+{
+	ModNameStorer mod_name_storer(getStack(), mod_name);
+
+	const std::string *init_filename = getClient()->getModFile(mod_name + ":init.lua");
+	const std::string display_filename = mod_name + ":init.lua";
+	if(init_filename == NULL)
+		throw ModError("Mod:\"" + mod_name + "\" lacks init.lua");
+
+	verbosestream << "Loading and running script " << display_filename << std::endl;
+
+	lua_State *L = getStack();
+
+	int error_handler = PUSH_ERROR_HANDLER(L);
+
+	bool ok = ScriptApiSecurity::safeLoadFile(L, init_filename->c_str(), display_filename.c_str());
+	ok = ok && !lua_pcall(L, 0, 0, error_handler);
```

Comment:
This may be getting a bit too clever (I don't think you originally wrote this, but still).  It might be better to expand this into an `if` block in order to make it more readable.
```C++
if (ok) {
        ok = !lua_pcall(L, 0, 0, error_handler);
}
```